relay-server/src/actors/processor.rs

```diff
@@ -2279,59 +2284,92 @@ impl EnvelopeProcessorService {
-            let validated_span = match self.validate_span(annotated_span) {
-                Ok(span) => span,
+        let received_at = state.managed_envelope.received_at();
+
+        state.managed_envelope.retain_items(|item| {
+            let annotated_span = match item.ty() {
+                ItemType::OtelSpan => {
+                    match serde_json::from_slice::<relay_spans::OtelSpan>(&item.payload()) {
+                        Ok(otel_span) => Annotated::new(otel_span.into()),
+                        Err(err) => {
+                            relay_log::debug!("failed to parse OTel span: {}", err);
+                            return ItemAction::DropSilently;
+                        }
+                    }
+                }
+                ItemType::Span => match Annotated::<Span>::from_json_bytes(&item.payload()) {
+                    Ok(span) => span,
+                    Err(err) => {
+                        relay_log::debug!("failed to parse span: {}", err);
+                        return ItemAction::DropSilently;
+                    }
+                },
+
+                _ => return ItemAction::Keep,
+            };
+
+            let config = NormalizeSpanConfig {
+                received_at,
+                max_secs_in_past: self.inner.config.max_secs_in_past(),
+                max_secs_in_future: self.inner.config.max_secs_in_future(),
+                max_tag_value_length: self
+                    .inner
+                    .config
+                    .aggregator_config_for(MetricNamespace::Spans)
+                    .max_tag_value_length,
+            };
+            let mut annotated_span = match normalize_span(annotated_span, config) {
+                Ok(s) => s,
                 Err(e) => {
-                    relay_log::error!("invalid span: {e}");
-                    return;
+                    relay_log::debug!("failed to normalize span: {}", e);
+                    return ItemAction::DropSilently;
                 }
             };
+
+            let Some(span) = annotated_span.value_mut() else {
+                return ItemAction::DropSilently;
+            };
+
             if let Some(config) = span_metrics_extraction_config {
-                if let Some(span_value) = validated_span.value() {
-                    let metrics = extract_metrics(span_value, config);
-                    state.extracted_metrics.project_metrics.extend(metrics);
-                }
-            }
-            if let Ok(payload) = validated_span.to_json() {
-                let mut item = Item::new(ItemType::Span);
-                item.set_payload(ContentType::Json, payload);
-                items.push(item);
+                let metrics = extract_metrics(span, config);
+                state.extracted_metrics.project_metrics.extend(metrics);
             }
-        };
-        let standalone_span_ingestion_enabled = state
-            .project_state
-            .has_feature(Feature::StandaloneSpanIngestion);
```

Comment:
Is the feature flag no longer useful? Without this check, we'll proceed with `ItemType::Span`s too.

Reply:
It's already checked in `span::filter` so we didn't think we needed to have it twice.